### ISSUE TITLE
Can't write Assert.xyz for a Map where the expected value of a key is 'null'

### ISSUE THREAD

**Issue by mictru** [cbeust/testng#2891]
### Expected behavior
The provided test case shouldn't fail - at least I think it should not fail.

### Actual behavior
The test case fails.

### Is the issue reproducible on runner?
- [x] NetBeans

### Test case sample
```
@Test
public void thisSucceeds() {
	var actual = new HashMap<String, String> ();
	actual.put("key", "1");
	var expected = new HashMap<String, String>();
	expected.put("key", "1");
	
	Assert.assertEqualsDeep(actual, expected);
}
@Test
public void thisFailsButShouldAlsoSucceed() {
	var actual = new HashMap<String, String> ();
	actual.put("key", null);
	var expected = new HashMap<String, String>();
	expected.put("key", null);
	
	Assert.assertEqualsDeep(actual, expected);
}
```

### Problem [Assert.assertEqualsDeep, Assert.java#L2063]
The method trys to retrieve the data type of the expected value `expectedValue.getClass()` which fails since the expected value of the key is null.

**Comment by krmahadevan** [comment]
The latest released version of TestNG is `7.7.1`. Please retry using that and let us know if the problem exists.

**Comment by mictru** [comment]
Sorry, cant test 7.7.1 since swapping the libs shipped with netbeans 17 (6.14.3) does not allow me to run any tests...

`Exception in thread "main" org.testng.TestNGException: 
Couldn't find a constructor in class org.testng.reporters.VerboseReporter
	at org.testng.internal.objects.InstanceCreator.newInstance(InstanceCreator.java:57)
	at org.testng.ITestObjectFactory.newInstance(ITestObjectFactory.java:10)
	at org.testng.TestNG.setListenerClasses(TestNG.java:679)
	at org.testng.TestNG.configure(TestNG.java:1519)
	at org.testng.TestNG.privateMain(TestNG.java:1405)
	at org.testng.TestNG.main(TestNG.java:1378)`

The method `assertEqualsDeep(Map<?, ?> actual, Map<?, ?> expected, String message)` has changed from 6.14.3 to 7.7.1 but both contain the `expectedValue.getClass()` call without checking that `expectedValue` is not null.
Method:
https://github.com/cbeust/testng/blob/89dc5845fcb46c26af187e50ea907a7382d06e72/testng-asserts/src/main/java/org/testng/Assert.java#L2143
https://github.com/cbeust/testng/blob/89dc5845fcb46c26af187e50ea907a7382d06e72/testng-asserts/src/main/java/org/testng/Assert.java#L2164


This method has the same problem, altough its much more unlikely that a set will contain a 'null' value...
https://github.com/cbeust/testng/blob/89dc5845fcb46c26af187e50ea907a7382d06e72/testng-asserts/src/main/java/org/testng/Assert.java#L2044
https://github.com/cbeust/testng/blob/89dc5845fcb46c26af187e50ea907a7382d06e72/testng-asserts/src/main/java/org/testng/Assert.java#L2063

**Comment by mictru** [comment]
I've created a maven test project with netbeans and added testng 7.7.1 and my testcase... and its running fine 😂 dont know why my ant-project throws exceptions at me when i swapped the libs, but maven works fine. and the testcase too.
`Objects.equals(...)` in 7.7.1 handles the map stuff (and also null values) while back in 6.13.4 there was just `actual == expected` so execution continues and reaches the problematic code section.

closed as not reproducible/fixed in 7.7.1

**Comment by krmahadevan** [comment]
@mictru - Thank you for the confirmation. Since you mention ant, am guessing that somewhere in your libs there perhaps is TestNG already available (and its an older version) or maybe you have a jar in your libs that contains a shaded TestNG jar within it. All these are just naive guesses since I have never used ant as a build tool.
